### PR TITLE
Add unarchiving support

### PR DESCRIPTION
**Pull request by Tsuesun**
The github API now allows unarchiving an archived repo. This removes the check so that we ar able to unarchive repos using pygithub.

https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#update-a-repository

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -2,3 +2,4 @@ cryptography
 httpretty>=1.0.3
 pytest>=5.3
 pytest-cov>=2.8
+pre-commit>=2.21.0
```
Comment (sfdye) [inline test-requirements.txt:5]: unrelated change, can you remove this?

**Review by sfdye — APPROVED** [review]

File: test-requirements.txt (hunk 1 above)
Comment (Tsuesun) [reply test-requirements.txt]: Done :+1:

**Comment by Tsuesun** [comment]
@sfdye, thanks for the review. Is there any particular issue you would like me to have a look at?

**Comment by EnricoMi** [comment]
> ... Is there any particular issue you would like me to have a look at?

I'd suggest to look at those issues / pull requests that have the most :+1:, that is a good indicator of popularity:
image
